lib/runner.js

```diff
@@ -36,14 +38,57 @@ const loadMigrationFiles = (db, options) =>
       throw new Error(`Can't get migration files: ${err.stack}`);
     });
 
+const lock = (db, options) => {
+  const schema = getMigrationTableSchema(options);
+  const lockName = crypto.randomBytes(30).toString('base64');
+  const getCurrentLockName = () =>
+    db
+      .select(`SELECT obj_description(c.oid) as "comment"
+                FROM pg_class c join pg_namespace n ON (c.relnamespace = n.oid)
+                WHERE c.relname = '${options.migrations_table}' and c.relkind = 'r' and n.nspname = '${schema}'`)
+      .then(rows => {
+        if (rows.length > 1) {
+          throw new Error('More then one migration table');
+        } else if (rows.length === 1) {
+          return rows[0]['comment'];
+        }
+        return null;
+      });
+
+  return getCurrentLockName()
+    .then(currentLockName => {
+      if (currentLockName) {
+        throw new Error('Another migration is already running');
+      }
+    })
+    .then(() => db.query(`COMMENT ON TABLE "${schema}"."${options.migrations_table}" IS '${lockName}'`))
+    .then(getCurrentLockName)
+    .then(currentLockName => {
+      if (currentLockName !== lockName) {
```

Comment:
Now when I'm looking on it. It is not 100% lock 🤦‍♂️ 
When some other `lock` manages to run between first `getCurrentLockName` and commenting on table, we will have two locks obtained...